conda_build_all/builder.py

```diff
@@ -107,8 +118,15 @@ def select_lines(data, *args, **kwargs):
             # is only a suitable solution when selectors are also comments.
             return data
 
+        meta.final = False
         with mock.patch('conda_build.metadata.select_lines', new=select_lines):
-            with mock.patch('conda_build.jinja_context.select_lines', new=select_lines):
+            try:
+                with mock.patch('conda_build.jinja_context.select_lines', new=select_lines):
+                    try:
+                        meta.parse_again(config, permit_undefined_jinja=True)
+                    except TypeError:
```

Comment:
What's going on here? How come sometimes a config is needed, and others it isn't?

Reply:
metadata has a config member.  Lots of cb2 passes around a config object unnecessarily.  cb3 cleans it up to use the one that's already there on the metadata object.